File: examples/java/src/main/java/jsontag/example/ShapeService.java
package jsontag.example;

import java.util.Random;

public class ShapeService {

    private final Random random = new Random();

    public Shape generateRandomShape() {
        int variant = random.nextInt(3);
        return switch (variant) {
            case 0 -> new Circle(getRandomShapeEdge());
            case 1 -> new Rectangle(getRandomShapeEdge(), getRandomShapeEdge());
            default -> new Triangle(getRandomShapeEdge(), getRandomShapeEdge(), getRandomShapeEdge());
        };
    }

    private int getRandomShapeEdge() {
        return random.nextInt(80, 121);
    }
}

File: examples/scala/src/main/scala/jsontag/example/ShapeService.scala
package jsontag.example

import scala.util.Random

object ShapeService:

  def generateRandomShape(): Shape =
    val variant = Random.nextInt(3)
    variant match
      case 0 => Circle(diameter = getRandomShapeEdge())
      case 1 => Rectangle(width = getRandomShapeEdge(), height = getRandomShapeEdge())
      case _ => Triangle(edgeA = getRandomShapeEdge(), edgeB = getRandomShapeEdge(), edgeC = getRandomShapeEdge())

  def getRandomShapeEdge(): Int =
    Random.between(80, 121)
